sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/SearchIndexClientBuilder.java

```diff
@@ -57,16 +109,54 @@
  * </pre>
  * <!-- end com.azure.search.documents.indexes.SearchIndexAsyncClient.instantiation -->
  *
- * <p><strong>Instantiating a synchronous Search Index Client</strong></p>
  *
- * <!-- src_embed com.azure.search.documents.indexes.SearchIndexClient.instantiation -->
+ * <h4>
+ *     Building and Authenticating a {@link SearchIndexClient} or {@link SearchIndexAsyncClient} using Azure Active Directory
+ * </h4>
+ *
+ * <p>
+ *   You can also create a {@link SearchIndexClient} or {@link SearchIndexAsyncClient} using Azure Active Directory (AAD)
+ *   authentication. Your user or service principal must be assigned the "Search Index Data Reader" role. Using the
+ *   DefaultAzureCredential you can authenticate a service using Managed Identity or a service principal, authenticate
+ *   as a developer working on an application, and more all without changing code. Please refer the <a href="https://learn.microsoft.com/en-us/azure/search/search-security-rbac?tabs=config-svc-portal,roles-portal,test-portal,custom-role-portal,disable-keys-portal">documentation</a> for
+ *   instructions on how to connect to Azure Cognitive Search using Azure role-based access control (Azure RBAC).
+ * </p>
+ *
+ * <p>
+ *     Before you can use the `DefaultAzureCredential`, or any credential type from Azure.Identity, you'll first need to install the Azure.Identity package.
+ * </p>
+ *
+ * <p>
+ *     To use DefaultAzureCredential with a client ID and secret, you'll need to set the `AZURE_TENANT_ID`, `AZURE_CLIENT_ID`,
+ *     and `AZURE_CLIENT_SECRET` environment variables; alternatively, you can pass those values to the
+ *     `ClientSecretCredential` also in azure-identity.
+ * </p>
+ *
+ * <p>
+ *     Make sure you use the right namespace for DefaultAzureCredential at the top of your source file:
+ * </p>
+ *
+ * <!-- src_embed DefaultAzureCredentialImports -->
  * <pre>
+ * import com.azure.identity.DefaultAzureCredential;
+ * import com.azure.identity.DefaultAzureCredentialBuilder;
+ * </pre>
+ * <!-- end DefaultAzureCredentialImports -->
+ *
+ * <p>
+ *     Then you can create an instance of DefaultAzureCredential and pass it to a new instance of your client:
+ * </p>
+ *
+ * <!-- src_embed com.azure.search.documents.SearchIndexClientBuilder.credential -->
```

Comment:
Should also add an asynchronous client creation example for parity with the API key section